src/lavinmq/queue/stream_queue.cr

```diff
@@ -0,0 +1,228 @@
+require "./durable_queue"
+require "../client/channel/consumer"
+
+module LavinMQ
+  class StreamQueue < Queue
+    @durable = true
+    @exclusive_consumer = false
+    @no_ack = true
+
+    private def init_msg_store(data_dir)
+      replicator = @vhost.@replicator
+      @msg_store = StreamQueueMessageStore.new(data_dir, replicator)
+    end
+
+    class StreamQueueMessageStore < MessageStore
+      @last_offset = 0_i64
+
+      def initialize(@data_dir : String, @replicator : Replication::Server?)
+        super
+      end
+
+      # Populate bytesize, size and segment_msg_count
+      private def load_stats_from_segments : Nil
```

Comment:
is this different from the base class implementation?

Reply:
ah, one line, updating the `@last_offset`, i think we should figure out a better way of doing that instead of copy pasting 20+ lines.